providers/implementations/serializers/deserialize_der2rsa.c

```diff
@@ -177,10 +199,27 @@ static int der2rsa_deserialize(void *vctx, OSSL_CORE_BIO *cin,
 
     if (rsa != NULL) {
         OSSL_PARAM params[3];
+        char *object_type = NULL;
+
+        switch (RSA_test_flags(rsa, RSA_FLAG_TYPE_MASK)) {
+        case RSA_FLAG_TYPE_RSA:
+            object_type = "RSA";
+            break;
+        case RSA_FLAG_TYPE_RSASSAPSS:
+            object_type = "RSA-PSS";
+            break;
+        default:
+            ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_RSA_KEY,
```

Comment:
Is there a reason why this is the only source file in providers that uses ERR_raise_data

Reply:
That's not true:

``` console
$ git grep -l ERR_raise_data providers
providers/implementations/rands/seeding/rand_vms.c
providers/implementations/signature/rsa.c
```

I guess that we all haven't found reason to add extra data that much, or simply haven't gotten used to it (the older `ERR_add_error_data` is klunky and quite discouraging to use!).

We do have an outstanding job to replace all the FOOerr() with ERR_raise(), and FOOerr() + ERR_add_error() with ERR_raise_data().  There are quite a few, but there is a script `util/err-to-raise` that's doing much of the job (thank you @richsalz) and that should be unleashed at some point.

Reply:
ProvErr() seems simpler to me.. oh well.

Reply:
The combination FOOerr() + ERR_add_error() seems simpler to you?????  I find it yucky.  But ok...